src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3003,7 +3003,7 @@ private static bool AreCloseEnough(Symbol original, Symbol updated)
         private static bool AreLambdaAndNewDelegateSimilar(LambdaSymbol l, NamedTypeSymbol n)
         {
             var invokeMethod = n.DelegateInvokeMethod;
-            return invokeMethod.Parameters.SequenceEqual(l.Parameters,
+            return invokeMethod!.Parameters.SequenceEqual(l.Parameters,
```

Comment:
>invokeMethod! [](http://example.com/codeflow?start=19&length=13)

Assert instead?

Reply:
An assert seems redundant since the dereference will fail at runtime regardless.

Reply:
> An assert seems redundant since the dereference will fail at runtime regardless.

I think we prefer using asserts to inform nullable analysis about expectations rather than suppressing diagnostics with ```!```.

Reply:
Chuck swayed me on this in some previous PR, but I don't think we circled back to Jared to update guidelines. If the code would clearly crash, then a suppression seems fine to me (assertion wouldn't add much).
Chuck, let's bring this up to team meeting or with Jared to confirm and capture.